### PR TITLE
Feature: create Order (POST /v1/orders)

### PR DESCRIPTION
**Pull request by andreagostinho-meli**
## 📝 Description

- Create Order client to make requests to the Order API
- Map Order request and response fields
- Add example of creating an Order with all possible fields
- Add integration and unit tests

### CODE REVIEW

**Review by danielen-meli — APPROVED** [review]

File: src/clients/order/create/types.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  capture_mode?: string;
+}
+
+export declare type TransactionsRequest = {
+  payments?: PaymentRequest[];
+}
+
+export declare type TransactionsResponse = {
+  payments?: PaymentResponse[];
+  refunds?: Refund[];
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:63]: Manter a nomenclatura
```suggestion
  refunds?: RefundResponse[];
```

File: src/clients/order/create/types.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+export declare type Item = {
+  title?: string;
+  unit_price?: string;
+  quantity?: number;
+  id?: string;
+  category_id?: string;
+  description?: string;
+  picture_url?: string;
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:151]:
```suggestion
  id?: string;
  title?: string;
  unit_price?: string;
  quantity?: number;
  category_id?: string;
  description?: string;
  picture_url?: string;
```

File: src/clients/order/create/types.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  type_config?: TypeConfig;
+  processing_mode?: string;
+  description?: string;
+  marketplace?: string;
+  marketplace_fee?: string;
+  items?: Item[];
+  expiration_time?: string;
+}
+
+export declare interface OrderResponse extends ApiResponse {
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:32]: Esta estruturas do tipo response podem ficar na pasta raiz /order dentro do arquivo commonTypes.ts, pois elas podem ser compartilhadas com os recursos como GET, UPDATE. Veja os recursos de payments para se basear

File: src/clients/order/create/types.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+export declare type TransactionsResponse = {
+  payments?: PaymentResponse[];
+  refunds?: Refund[];
+}
+
+export declare type PaymentRequest = {
+  amount?: string;
+  automatic_payments?: AutomaticPayments;
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:68]: Não localizei essa referencia

File: src/clients/order/create/types.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+  title?: string;
+  unit_price?: string;
+  quantity?: number;
+  id?: string;
+  category_id?: string;
+  description?: string;
+  picture_url?: string;
+}
+
+export declare type AutomaticPayments = {
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:154]:
```suggestion
export declare type AutomaticPaymentsRequest = {
```

File: src/clients/order/create/types.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+  payment_profile_id?: string;
+  retries?: number;
+  schedule_date?: string;
+  due_date?: string;
+}
+
+export declare type StoredCredential = {
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:161]:
```suggestion
export declare type StoredCredentialRequest = {
```

File: src/clients/order/create/types.ts [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+  payment_initiator?: string;
+  reason?: string;
+  store_payment_method?: boolean;
+  first_payment?: boolean;
+}
+
+export declare type SubscriptionData = {
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:168]:
```suggestion
export declare type SubscriptionDataRequest = {
```

File: src/clients/order/create/types.ts [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  type?: string;
+  number?: string;
+}
+
+export declare type Address = {
+  street_name?: string;
+  street_number?: string;
+}
+
+export declare type Item = {
```
Comment (eltinMeli) [inline src/clients/order/create/types.ts:144]: Migrar para commonType

File: src/clients/order/create/types.ts (hunk 4 above)
Comment (andreagostinho-meli) [reply src/clients/order/create/types.ts:68]: Aqui: https://github.com/andreagostinho-meli/sdk-nodejs/pull/1/files#diff-c826b78af85621a01f3432a68c9c693cca1a5b5a85dc3c5ae275fc1e020df662R154

**Pull request merged**